src/component/DefaultTooltipContent.tsx

```diff
@@ -63,56 +64,107 @@ export class DefaultTooltipContent<TValue extends ValueType, TName extends NameT
   };
 
   renderContent() {
-    const { payload, separator, formatter, itemStyle, itemSorter } = this.props;
+    const {
+      payload,
+      separator,
+      formatter,
+      itemStyle,
+      itemSorter,
+      tooltipDatakey,
+      labelStyle,
+      labelFormatter,
+      labelClassName,
+    } = this.props;
+
+    const finalLabelStyle = {
+      margin: 0,
+      ...labelStyle,
+    };
 
     if (payload && payload.length) {
       const listStyle = { padding: 0, margin: 0 };
 
-      const items = (itemSorter ? _.sortBy(payload, itemSorter) : payload).map((entry, i) => {
-        if (entry.type === 'none') {
-          return null;
-        }
-
-        const finalItemStyle = {
-          display: 'block',
-          paddingTop: 4,
-          paddingBottom: 4,
-          color: entry.color || '#000',
-          ...itemStyle,
-        };
-        const finalFormatter = entry.formatter || formatter || defaultFormatter;
-        let { value, name } = entry;
-        if (finalFormatter && value != null && name != null) {
-          const formatted = finalFormatter(value, name, entry, i, payload);
-          if (Array.isArray(formatted)) {
-            [value, name] = formatted as [TValue, TName];
-          } else {
-            value = formatted;
-          }
-        }
-        return (
-          // eslint-disable-next-line react/no-array-index-key
-          <li className="recharts-tooltip-item" key={`tooltip-item-${i}`} style={finalItemStyle}>
-            {isNumOrStr(name) ? <span className="recharts-tooltip-item-name">{name}</span> : null}
-            {isNumOrStr(name) ? <span className="recharts-tooltip-item-separator">{separator}</span> : null}
-            <span className="recharts-tooltip-item-value">{value}</span>
-            <span className="recharts-tooltip-item-unit">{entry.unit || ''}</span>
-          </li>
-        );
-      });
-
-      return (
-        <ul className="recharts-tooltip-item-list" style={listStyle}>
-          {items}
-        </ul>
-      );
+      const run = _.flow([
+        // Group payload by datakey value
+        payloadElement => {
+          return _.groupBy(payloadElement, el => el?.payload[tooltipDatakey]);
+        },
+
+        labelGroupedPayload => {
+          return _.map(labelGroupedPayload, (payloadPayload, payloadLabel) => {
+            // Build label from datakey value
+            const hasLabel = !_.isNil(payloadLabel);
+            let finalLabel: ReactNode = hasLabel ? payloadLabel : '';
+            const labelCN = classNames('recharts-tooltip-label', labelClassName);
+
+            if (hasLabel && labelFormatter && payload !== undefined && payloadPayload !== null) {
```

Comment:
logic was already here but can use loose equals to cover both undefined and null in one go
```suggestion
            if (hasLabel && labelFormatter && payloadPayload != null) {
```